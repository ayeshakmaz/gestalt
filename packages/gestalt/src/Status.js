// @flow strict
import React, { type Node } from 'react';
import PropTypes from 'prop-types';
import Icon from './Icon.js';
import Tooltip from './Tooltip.js';
// import styles from './Status.css';

export type StatusColor = 'gray' | 'green' | 'orange' | 'red';

type Props = {|
  type?: string,
|};

export default function Status(props: Props): Node {
  const { type } = props;

  let accessibilityLabel;
  let color;
  let icon;

  switch (type) {
    case 'unstarted':
      icon = 'workflow-status-unstarted';
      color = 'darkGray';
      accessibilityLabel = 'Unstarted';
      break;
    case 'in-progress':
      icon = 'workflow-status-in-progress';
      color = 'green';
      accessibilityLabel = 'Unstarted';
      break;
    case 'halted':
      icon = 'workflow-status-halted';
      color = 'darkGray';
      accessibilityLabel = 'Unstarted';
      break;
    case 'ok':
      icon = 'workflow-status-ok';
      color = 'green';
      accessibilityLabel = 'Unstarted';
      break;
    case 'problem':
      icon = 'workflow-status-problem';
      color = 'red';
      accessibilityLabel = 'Unstarted';
      break;
    /*
    case 'canceled':
      icon = 'workflow-status-canceled';
      color = 'green';
      accessibilityLabel = 'Unstarted';
      break;
    */
    case 'warning':
      icon = 'workflow-status-warning';
      color = 'orange';
      accessibilityLabel = 'Unstarted';
      break;
    default:
      icon = 'workflow-status-unstarted';
      color = 'darkGray';
      accessibilityLabel = 'Unstarted';
  }

  return (
    <Tooltip text="Send Pin">
      <Icon accessibilityLabel={accessibilityLabel} color={color} icon={icon} />
    </Tooltip>
  );
}

Status.propTypes = {
  type: PropTypes.string,
};
