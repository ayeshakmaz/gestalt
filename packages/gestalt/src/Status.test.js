// @flow strict
import React from 'react';
import { create } from 'react-test-renderer';
import Status from './Status.js';

describe('Status', () => {
  it('renders', () => {
    const tree = create(<Status type="unstarted" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
