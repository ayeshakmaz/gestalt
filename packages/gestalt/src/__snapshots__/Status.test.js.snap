// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders 1`] = `
<svg
  aria-hidden={null}
  aria-label="foo"
  className="icon red iconBlock"
  height={16}
  role="img"
  viewBox="0 0 24 24"
  width={16}
>
  <path
    d="test-file-stub"
  />
</svg>
`;

exports[`Status renders an accessibility label 1`] = `
<svg
  aria-hidden={null}
  aria-label="foo"
  className="icon red iconBlock"
  height={16}
  role="img"
  viewBox="0 0 24 24"
  width={16}
>
  <path
    d="test-file-stub"
  />
</svg>
`;
